_docs/overview/attack-surface.md

```diff
@@ -4,65 +4,100 @@ description: Learn about the attack survace for Glacier, the
+Scenarios in which a critical failure might happen. Possibly, but not necessarily, due
+to a malware infection.
+
+### Exfiltration of critically sensitive data
+A Quarantined Computer could leak critically sensitive data over a side channel, which
+is an attack based on information gained from the implementation of a computer system,
+rather than weaknesses in the implemented algorithm itself. Complementary malware on a 
+networked or attacker-controlled device in range then steals the data. Possibilities include:
+  * **Visual side channel**: Exploit of sensitive visual data. Data is displayed on the
+  screen as part of the protocol, so does not require malware on the quarantined computer.
+  If the protocol is followed, the attack surface here should be narrow, as users are
+  instructed to block all visual side channels. However, at a minimum, a smartphone with a
+  camera is used for reading QR codes.
+  * **Acoustic side channel**: Exploit of sounds emitted by computer or other device, if
+  inadequately blocked by sound insulation or masking noise. [Data theft by fan noise](https://www.wired.com/2016/06/clever-attack-uses-sound-computers-fan-steal-data), for example.
+  * **Radio side channel**: [Exploit of electromagnetic emanations from wired and wireless
+  keyboards](https://www.usenix.org/legacy/event/sec09/tech/full_papers/vuagnoux.pdf),
+  [transmission of radio signals from monitor to FM radio receiver](http://cyber.bgu.ac.il/content/how-leak-sensitive-data-isolated-computer-air-gap-near-mobile-phone-airhopper)
+  or [radio waves leaked by a processor's power use](https://www.wired.com/2015/06/radio-bug-can-steal-laptop-crypto-keys-fits-inside-pita/),
+  for example.
+  * **Seismic side channel**: Decoding vibrations from nearby keyboard using mobile phone accelerometer, for example.
+  * **Thermal side channel**: [Exploit using thermal sensors to detect heat emission](http://cyber.bgu.ac.il/blog/bitwhisper-heat-air-gap), for example.
+  * **Magnetic side channel**: [Exploit using smartphone magnetic field sensors to extract hard drive data](http://fc15.ifca.ai/preproceedings/paper_14.pdf), for example.
+  * **Rogue QR code**: Malware on a Quarantined Computer could exfiltrate critically
+  sensitive data via QR codes IF cooperating malware on the QR reading device is prepared
+  to steal the data. The risk of this scenario is negligible; unless the attacker
+  simultaneously compromised every major smartphone QR reader with cooperating malware,
+  any manipulation of QR codes would be quickly detected by users of non-compromised QR
+  reader software, leading to widespread awareness and isolation of the threat. This makes
+  it a very unattractive attack vector.
+  * **Data leak**: Critically sensitive data could be leaked, intentionally or otherwise, as part of the
+  payload of valid data. For example, the nonce used for a transaction signature could contains
```

Comment:
"contain" not "contains"

Reply:
Corrected in a new commit.intentionally or otherwise, as part of

Reply:
* Corrected in a new commit.